lib/shard/src/segment_holder/mod.rs

```diff
@@ -978,6 +990,40 @@ impl SegmentHolder {
         Ok(LockedSegment::new(segment))
     }
 
+    /// Method tries to remove the segment with the given ID under the following conditions:
+    ///
+    /// - The segment exists in the holder, if not - it is ignored.
+    /// - The segment is a raw segment and not some special proxy segment.
```

Comment:
> The segment is a raw segment and not some special proxy segment.

We don't check this condition.

We remove the segment from the holder, which succeeds no matter the type of segment. After that we simply add it again or drop the data, not checking the segment type.